components/webrender_traits/lib.rs

```diff
@@ -115,25 +126,45 @@ impl webrender_api::ExternalImageHandler for WebrenderExternalImageHandlers {
         let handler_type = external_images
             .get(&key)
             .expect("Tried to get unknown external image");
-        let (texture_id, uv) = match handler_type {
+        match handler_type {
             WebrenderImageHandlerType::WebGL => {
-                let (texture_id, size) = self.webgl_handler.as_mut().unwrap().lock(key.0);
-                (
-                    texture_id,
-                    TexelRect::new(0.0, size.height as f32, size.width as f32, 0.0),
-                )
+                let (source, size) = self.webgl_handler.as_mut().unwrap().lock(key.0);
+                let texture_id = match source {
+                    WebrenderImageSource::TextureHandle(b) => b,
+                    _ => panic!("Wrong type"),
+                };
+                webrender_api::ExternalImage {
+                    uv: TexelRect::new(0.0, size.height as f32, size.width as f32, 0.0),
+                    source: webrender_api::ExternalImageSource::NativeTexture(texture_id),
+                }
             },
             WebrenderImageHandlerType::Media => {
-                let (texture_id, size) = self.media_handler.as_mut().unwrap().lock(key.0);
-                (
-                    texture_id,
-                    TexelRect::new(0.0, 0.0, size.width as f32, size.height as f32),
-                )
+                let (source, size) = self.media_handler.as_mut().unwrap().lock(key.0);
+                let texture_id = match source {
+                    WebrenderImageSource::TextureHandle(b) => b,
+                    _ => panic!("Wrong type"),
+                };
+                webrender_api::ExternalImage {
+                    uv: TexelRect::new(0.0, size.height as f32, size.width as f32, 0.0),
+                    source: webrender_api::ExternalImageSource::NativeTexture(texture_id),
+                }
+            },
+            WebrenderImageHandlerType::WebGPU => {
+                let (source, size) = self.webgpu_handler.as_mut().unwrap().lock(key.0);
+                let buffer = match source {
+                    WebrenderImageSource::Raw(b) => b,
+                    _ => panic!("Wrong type"),
+                };
+                webrender_api::ExternalImage {
+                    uv: TexelRect::new(0.0, size.height as f32, size.width as f32, 0.0),
+                    source: webrender_api::ExternalImageSource::RawData(
```

Comment:
does it have to have a raw pointer in here? can we store `Vec` in it?

Reply:
I have figured this one out and am storing in a `Vec` now. I am cloning the vector from `PresentationData` and storing it in a hashmap in `locked_ids`  and forwarding it's slice to webrender. Haven't pushed those changes yet.